internal/tba/tba.go

```diff
@@ -0,0 +1,127 @@
+package tba
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/Pigmice2733/peregrine-backend/internal/store"
+)
+
+// Service provides an interface to retreive data from
+// The Blue Alliance's API
+type Service struct {
+	URL    string
+	APIKey string
+}
+
+type tbaDistrict struct {
+	Abbreviation string `json:"abbreviation"`
+}
+
+type tbaWebcast struct {
+	Type    string `json:"type"`
+	Channel string `json:"channel"`
+}
+
+type tbaEvent struct {
+	Key          string       `json:"key"`
+	ShortName    string       `json:"short_name"`
+	District     *tbaDistrict `json:"district"`
+	Lat          float64      `json:"lat"`
+	Lng          float64      `json:"lng"`
+	LocationName string       `json:"location_name"`
+	Week         *int         `json:"week"`
+	StartDate    string       `json:"start_date"`
+	EndDate      string       `json:"end_date"`
+	TimeZone     string       `json:"timezone"`
+	Webcasts     []tbaWebcast `json:"webcasts"`
+}
+
+func (s *Service) makeRequest(path string) (*http.Response, error) {
+	req, err := http.NewRequest("GET", s.URL+path, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	req.Header.Set("X-TBA-Auth-Key", s.APIKey)
+
+	return http.DefaultClient.Do(req)
+}
+
+func webcastURL(webcastType string, channel string) (string, store.WebcastType) {
+	if string(store.Twitch) == webcastType {
+		return fmt.Sprintf("https://www.twitch.tv/%v", channel), store.Twitch
+	} else if string(store.Youtube) == webcastType {
+		return fmt.Sprintf("https://www.youtube.com/watch?v=%v", channel), store.Youtube
+	}
+	return "", ""
+}
+
+// GetEvents retreives all events from the given year (e.g. 2018).
+func (s *Service) GetEvents(year int) ([]store.Event, error) {
+	path := fmt.Sprintf("/events/%v", year)
+
+	response, err := s.makeRequest(path)
+	if err != nil {
+		return nil, fmt.Errorf("TBA request failed: %v", err)
```

Comment:
I think you should go with errors.Wrap for all these error messages.
`errors.Wrap(err, "cannot complete request to the blue alliance api")`